### PR TITLE
luvit: fix ARM and Linux build

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- luvit: fix ARM build
- luv: switch to upstream LuaJIT
- neovim: switch to upstream LuaJIT

The ARM build for `luvit` is broken by (1) a LuaJIT bug and (2) outdated linker flags in the `luvi` build.

The LuaJIT bug is fixed in upstream LuaJIT, so let's use that. We also use a newer version of `luvi` in order to avoid the outdated linker flags. I've opted to update `luvi` instead of using a patch because the new version of `luvi` is also the one that matches our versions of `luajit` and `luv`.

To avoid potential risks of binary incompatibility, let's use upstream LuaJIT in `luv` and `neovim` too. `luv` is a dependency of `luvit`, and `neovim` depends on `luv`. The risks are admittedly relatively small, but it's probably better to play it safe here.

Finally, let's also avoid static linkage when building `luvit`, since that's more in line with what we do everywhere else.

### CODE REVIEW

**Review by SMillerDev — DISMISSED** [review]

File: Formula/luvit.rb [hunk 1]
```diff
@@ -32,13 +33,14 @@ class Luvit < Formula
   # https://github.com/luvit/lit/raw/$(LIT_VERSION)/get-lit.sh
   resource "luvi" do
     url "https://github.com/luvit/luvi.git",
-        tag:      "v2.12.0",
-        revision: "5d1052f11e813ff9edc3ec75b5282b3e6cb0f3bf"
+        tag:      "v2.14.0", # Install a newer version to fix ARM64.
+        revision: "450ec116aef6f8cb0cba83c7e61c9bce112d6deb"
   end
```
Comment (SMillerDev) [inline Formula/luvit.rb:38]: If this works, can we make a PR upstream for it?

Comment (carlocab) [reply Formula/luvit.rb:38]: Do you just mean to update the `luvi` version in `get-lit.sh` (which is what we reference for the `luvi` version)?

Comment (SMillerDev) [reply Formula/luvit.rb:38]: Yeah, if it works for us it could be worth consideration for upstream

**Comment by carlocab** [comment]
Nope, doesn't fix ARM yet. Not sure why; I did some testing with an earlier iteration of this PR on ARM and it worked...

**Comment by danielnachun** [comment]
Was this the failure that was supposed to be fixed on ARM?  Because that seems to be why it's failing.
```
 fail: [string "bundle:libs/core.lua"]:234: Problem downloading custom luvi: https://github.com/luvit/luvi/releases/download/v2.12.0/luvi-regular-Darwin_arm64
```
Is there a way to disable it from trying to download its own copy of `luvi`, given that we supposed to be using our own instead?

**Comment by cho-m** [comment]
> Nope, doesn't fix ARM yet. Not sure why; I did some testing with an earlier iteration of this PR on ARM and it worked...

Were you testing with patched copy of `luvi` 2.12.0?

I think `lit` will try to match `luvi` version from https://github.com/luvit/luvit/blob/2.18.1/package.lua#L4-L5
```
  luvi = {
    version = "2.12.0",
```

As local version is now 2.14.0, it looks like it is trying to download a non-existent pre-built of 2.12.0
```
downloading: https://github.com/luvit/luvi/releases/download/v2.12.0/luvi-regular-Darwin_arm64
```

**Comment by carlocab** [comment]
> Was this the failure that was supposed to be fixed on ARM?

No, the failure that needed to be fixed on ARM was the linker getting upset over LuaJIT generating malformed bytecode on ARM macOS.

> > Nope, doesn't fix ARM yet. Not sure why; I did some testing with an earlier iteration of this PR on ARM and it worked...
> 
> Were you testing with patched copy of `luvi` 2.12.0?
> 
> I think `lit` will try to match `luvi` version from https://github.com/luvit/luvit/blob/2.18.1/package.lua#L4-L5
> 
> ```
>   luvi = {
>     version = "2.12.0",
> ```
> 
> As local version is now 2.14.0, it looks like it is trying to download a non-existent pre-built of 2.12.0
> 
> ```
> downloading: https://github.com/luvit/luvi/releases/download/v2.12.0/luvi-regular-Darwin_arm64
> ```

Yea, probably. I thought I tested this version of the PR yesterday. Let's try patching 2.12 instead.

**Comment by carlocab** [comment]
Ok, that works. Thanks @cho-m.

**Comment by carlocab** [comment]
Incidentally, this fixes the Linux build of `luvit` too. But that's probably from linking dynamically.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**